<?xml version="1.0"?>
<launch>
    <arg name="stop_line_detector_settings" default="/home/amsl/catkin_ws/src/rwrc23/config/param/stop_line_detector/ikuta_outdoor.yaml"/>
    <!-- <arg name="stop_line_detector_settings" default="/home/amsl/catkin_ws/src/rwrc23/config/param/stop_line_detector/tsukuba_sunny.yaml"/> -->
    <!-- <arg name="stop_line_detector_settings" default="/home/amsl/catkin_ws/src/rwrc23/config/param/stop_line_detector/tsukuba_cloudy.yaml"/> -->
    <arg name="rviz_settings" default="$(find rwrc23_stop_line_detector)/config/rviz/stop_line_detector.rviz"/>
    <node pkg="rwrc23_stop_line_detector" type="stop_line_detector.py" name="stop_line_detector" output="screen">
        <rosparam command="load" file="$(arg stop_line_detector_settings)"/>
    </node>

    <!-- play rosbag -->
    <!-- Tsukuba sunny -->
    <!-- <arg name="bagfile" default="/share/share/RWRC/rwrc22/tsukuba/tsukuba_whiteline1.bag"/> -->
    <!-- <arg name="bagfile" default="/share/share/RWRC/rwrc22/tsukuba/iki1_with_realsense_2022-11-06-11-01-34.bag"/> -->
    <!-- Tsukuba cloudy -->
    <!-- <arg name="bagfile" default="/share/share/RWRC/rwrc23/bagfiles/tsukuba_23-07-15/stopline_data/stopline_data_2023-07-15-14-48-27.bag"/> -->
    <!-- <arg name="bagfile" default="/share/share/RWRC/rwrc23/bagfiles/tsukuba_23-07-15/stopline_data/stopline_data_2023-07-15-14-50-23.bag"/> -->
    <!-- <arg name="bagfile" default="/share/share/RWRC/rwrc23/bagfiles/tsukuba_23-07-15/stopline_data/stopline_data_2023-07-15-14-56-20.bag"/> -->
    <!-- <arg name="bagfile" default="/share/share/RWRC/rwrc23/bagfiles/tsukuba_23-07-15/stopline_data/stopline_data_2023-07-15-14-59-38.bag"/> -->

    <!-- Ikuta -->
    <!-- <arg name="bagfile" default="/home/amsl/bagfiles/bagfiles/rwrc/2023/ikuta_stopline1_2023-09-16-15-20-23.bag"/> -->
    <!-- <arg name="bagfile" default="/home/amsl/bagfiles/bagfiles/rwrc/2023/ikuta_stopline2_2023-09-16-15-20-56.bag"/> -->
    <!-- <arg name="bagfile" default="/home/amsl/bagfiles/bagfiles/rwrc/2023/ikuta_stopline3_2023-09-16-15-22-22.bag"/> -->
    <arg name="bagfile" default="/home/amsl/bagfiles/bagfiles/rwrc/2023/ikuta_stopline4_2023-09-16-15-23-42.bag"/>

    <arg name="start" default="0"/>
    <arg name="duration" default="200"/>
    <arg name="delay" default="30.0"/>
    <arg name="play_speed_ratio" default="1.0"/>
    <node pkg="rosbag" type="play" name="rosbag_play" output="screen" args="
        $(arg bagfile)
        -l
        -r $(arg play_speed_ratio)
        -s $(arg start)
        --clock
        ">
    </node>
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(arg rviz_settings)"/>
    <node name="marker_publisher" pkg="rostopic" type="rostopic"
        args="pub /detect_line std_msgs/Bool -r 10 true"/>

 <!-- extract images from a movie -->
    <!-- <node name="republish" type="republish" pkg="image_transport" output="screen" -->
    <!--     args="compressed in:=/stop_line_image raw out:=/stop_line_image" /> -->
    <!-- <node name="extract" pkg="image_view" type="extract_images" respawn="false" output="screen" cwd="ROS_HOME"> -->
    <!--     <remap from="image" to="/stop_line_image"/> -->
    <!-- </node> -->

</launch>
